src/linker/Linker/LinkContext.cs

```diff
@@ -394,37 +387,50 @@ public void SetAction (AssemblyDefinition assembly, AssemblyAction defaultAction
 			Annotations.SetAction (assembly, action);
 		}
 #endif
-		public AssemblyAction CalculateAssemblyAction (AssemblyNameDefinition name)
+		public AssemblyAction CalculateAssemblyAction (AssemblyDefinition assembly)
 		{
-			if (_actions.TryGetValue (name.Name, out AssemblyAction action))
+			if (_actions.TryGetValue (assembly.Name.Name, out AssemblyAction action))
 				return action;
 
-			if (IsCore (name))
-				return CoreAction;
+			if (CheckIsTrimmable (assembly))
+				return TrimAction;
 
-			return UserAction;
+			return DefaultAction;
 		}
 
-		public static bool IsCore (AssemblyNameReference name)
+		bool CheckIsTrimmable (AssemblyDefinition assembly)
 		{
-			switch (name.Name) {
-			case "mscorlib":
-			case "Accessibility":
-			case "Mono.Security":
-			// WPF
-			case "PresentationFramework":
-			case "PresentationCore":
-			case "WindowsBase":
-			case "UIAutomationProvider":
-			case "UIAutomationTypes":
-			case "PresentationUI":
-			case "ReachFramework":
-			case "netstandard":
-				return true;
-			default:
-				return name.Name.StartsWith ("System")
-					|| name.Name.StartsWith ("Microsoft");
+			if (!assembly.HasCustomAttributes)
+				return false;
+
+			bool isTrimmable = false;
+			var assemblyFileName = Resolver.GetAssemblyFileName (assembly);
+
+			foreach (var ca in assembly.CustomAttributes) {
+				if (!ca.AttributeType.IsTypeOf<AssemblyMetadataAttribute> ())
+					continue;
+
+				var args = ca.ConstructorArguments;
+				if (args.Count != 2)
+					continue;
+
+				if (args[0].Value is not string key || !key.Equals ("IsTrimmable", StringComparison.OrdinalIgnoreCase))
+					continue;
+
+				if (args[1].Value is not string value || !value.Equals ("True", StringComparison.OrdinalIgnoreCase)) {
+					LogWarning ($"Invalid AssemblyMetadata(\"IsTrimmable\", \"{args[1].Value}\") attribute in assembly '{assembly.Name.Name}'. Value must be \"True\"", 2102, assemblyFileName);
+					continue;
+				}
+
+				if (isTrimmable) {
+					LogWarning ($"Duplicate AssemblyMetadata(\"IsTrimmable\", \"True\") attributes in assembly '{assembly.Name.Name}'", 2103, assemblyFileName);
```

Comment:
Not sure we should be that pedantic

Reply:
I removed the warning about duplicate attributes